[Pipelines] Add a section on generating captions and embeddings for Pix2Pix Zero (#2395)
huggingface/diffusers: docs/source/en/api/pipelines/stable_diffusion/pix2pix_zero.mdx

    * Swap the `source_embeds` and `target_embeds`.
    * Change the input prompt to include "dog".  
* To learn more about how the source and target embeddings are generated, refer to the [original 

> paper](https://arxiv.org/abs/2302.03027). Below, we also provide a some directions on how to generate the embeddings.

Reviewer: ```suggestion
paper](https://arxiv.org/abs/2302.03027). Below, we also provide some directions on how to generate the embeddings.
```

_Coming soon_

## Generating source and target embeddings 

The authors originally used the [GPT-3 API](https://openai.com/api/) to generate the source and target captions for discovering

> edit directions. However, we can also leverage the open-source and the publicly models for this purpose.

Reviewer: ```suggestion
edit directions. However, we can also leverage open source and public models for the same purpose.
```

Below, we provide an end-to-end example with the [Flan-T5](https://huggingface.co/docs/transformers/model_doc/flan-t5) model
for generating captions and [CLIP](https://huggingface.co/docs/transformers/model_doc/clip) for
computing embeddings on the generated captions.  

**1. Load the generation model**:

```py
import torch
from transformers import AutoTokenizer, T5ForConditionalGeneration

tokenizer = AutoTokenizer.from_pretrained("google/flan-t5-xl")
model = T5ForConditionalGeneration.from_pretrained("google/flan-t5-xl", device_map="auto", torch_dtype=torch.float16)
```

**2. Construct a starting prompt**: 

```py
source_concept = "cat"
target_concept = "dog"

source_text = f"Provide a caption for images containing a {source_concept}. "
"The captions should be in English and should be no longer than 150 characters."

> target_text = source_text.replace(source_concept, target_concept)

Reviewer: nit: I'd repeat the prompt construction instead of doing a string replace, the user may inadvertently replace other parts of the prompt if they change the example concepts to something else.

```

Here, we're interested in the "cat -> dog" direction. 

**3. Generate captions**:

We can use a utility like so for this purpose. 

```py
def generate_captions(input_prompt):
    input_ids = tokenizer(input_prompt, return_tensors="pt").input_ids.to("cuda")

    outputs = model.generate(
        input_ids, temperature=0.8, num_return_sequences=16, do_sample=True, max_new_tokens=128, top_k=10
    )
    return tokenizer.batch_decode(outputs, skip_special_tokens=True)
```

> And the just call:

Reviewer: ```suggestion
And then we just call it to generate our captions:
```
